### PR TITLE
Vet-53 Sign up function and User database

### PR DESCRIPTION
**Pull request by maclipski**
Added user database and all necessary routes. For now only sign up function works, login will be added during next sprint. Sign up is already configured - requires unique email and encrypts passwords with Salt.

### CODE REVIEW

File: backend/routes/user.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const express = require("express")
+
+// controller functions
+const { signupUser, loginUser} = require("../controllers/userController")
```
Comment (ghost) [inline backend/routes/user.js]: Please remove extra blank space or add symetric one

File: backend/models/userModel.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        unique: true
+    },
+    password: {
+        type: String, 
+        require: true
+    }
+})
+
+// static signup method
+userSchema.statics.signup = async function( email, password ) {
```
Comment (ghost) [inline backend/models/userModel.js]: Please keep to one style throughout code (arg, arg) or ( arg, arg )

File: backend/models/userModel.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    const salt = await bcrypt.genSalt(10)
+    const hash = await bcrypt.hash(password, salt)
+
+    const user = await this.create({email, password: hash})
+
+    return user
+}
+
+module.exports = mongoose.model('User', userSchema)
```
Comment (ghost) [inline backend/models/userModel.js:37]: Please keep to one style throughout code (arg, arg) or ( arg, arg )

File: backend/models/userModel.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    password: {
+        type: String, 
+        require: true
+    }
+})
+
+// static signup method
+userSchema.statics.signup = async function( email, password ) {
+
+    const exists = await this.findOne({ email })
```
Comment (ghost) [inline backend/models/userModel.js]: Also here comment about one style of brackets

File: backend/models/userModel.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    if (exists) {
+        throw Error('Email already in use')
+    }
+
+    const salt = await bcrypt.genSalt(10)
```
Comment (ghost) [inline backend/models/userModel.js:28]: Could you please add a comment explaining what is salt f.e. ("A random string of characters added to the hash as another layer of protection.")

**Review by ghost — COMMENTED** [review]
Decorative changes

File: backend/models/userModel.js (hunk 4 above)
Comment (maclipski) [reply backend/models/userModel.js]: Unified all previous comments to (arg, arg) option

File: backend/models/userModel.js (hunk 5 above)
Comment (maclipski) [reply backend/models/userModel.js:28]: As requested three explaining comments added to the encrypting function

**Review by krawczykaghust — APPROVED** [review]
Approved

**Pull request merged**